File: esbmc.cpp/validador_titulo/validador_nao_comecar_com_numero_ou_caractere_especial.cpp
#include <cctype>

#define MAX_LEN_TITLE 21

using namespace std;

bool validate_no_start_with_number_or_special(const char title[MAX_LEN_TITLE]) {
    const char specials[] = "!@#$%^&*()_+=-[]{}|;:,.<>?/~`";
    __ESBMC_assert(!isdigit(title[0]), "O título não pode começar com um número ou caractere especial.");
    for (int j = 0; specials[j] != '\0'; j++) {
        __ESBMC_assert(title[0] != specials[j], "O título não pode começar com um número ou caractere especial.");
    }
    return true;
}

int main() {
    char title[MAX_LEN_TITLE];

    validate_no_start_with_number_or_special(title);

    return 0;
}

File: esbmc.py/Testes02/validador_titulo/validador_nao_comecar_com_numero_ou_caractere_especial.py
MAX_LEN_TITLE: int = 21

def validate_no_start_with_number_or_special(title: str, length: int) -> bool:
    specials: str = "!@#$%^&*()_+=-[]{}|;:,.<>?/~`"

    assert not ('0' <= title[0] <= '9'), \
        "O título não pode começar com um número ou caractere especial."

    
    j: int = 0
    while j < len(specials):
        assert title[0] != specials[j], \
            "O título não pode começar com um número ou caractere especial."
        j += 1

    return True

def main() -> None:
   
    title: str = input()
    length: int = len(title)

    assert 1 <= length <= 20 

    validate_no_start_with_number_or_special(title, length)

main()